src/com/amplitude/api/AmplitudeClient.java

```diff
@@ -703,6 +701,18 @@ public void run() {
             } else if (stringResponse.equals("request_db_write_failed")) {
                 Log.w(TAG,
                         "Couldn't write to request database on server, will attempt to reupload later");
+            } else if (response.code() == 413) {
+                // If blocked by one massive event, drop it
+                DatabaseHelper dbHelper = DatabaseHelper.getDatabaseHelper(context);
+                if (eventUploadMaxBatchSize == 1) {
+                    dbHelper.removeEvents(maxId);
```

Comment:
It shouldn't since eventUploadMaxBatchSize = 1. It will only fetch 1 event from the dbHelper, which should be the first event. I verified this behavior with my tests and through debugging. I did it this way since I saw the javascript code doing this, although if you want I can explicitly fetch the 0th event and just remove that.